eZ/Publish/API/Repository/Tests/RoleServiceTest.php

```diff
@@ -1872,6 +1873,81 @@ public function testGetRoleAssignmentsContainExpectedLimitation(array $roleAssig
         );
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\RoleService::loadRoleAssignments()
+     */
+    public function testLoadRoleAssignments(): void
+    {
+        $repository = $this->getRepository();
+        $roleService = $repository->getRoleService();
+
+        $role = $this->createRoleWithPolicies('testLoadRoleAssignments', []);
+        $user = $this->createUser('test', 'Test', 'Test');
+        $user2 = $this->createUser('test2', 'Test2', 'Test2');
+
+        $roleService->assignRoleToUser($role, $user);
+        $roleService->assignRoleToUser($role, $user2);
+
+        $loadedRole = $roleService->loadRole($role->id);
+
+        $roleAssignments = $roleService->loadRoleAssignments($loadedRole, 0, 1);
+
+        self::assertCount(1, $roleAssignments);
+        self::assertInstanceOf(UserRoleAssignment::class, $roleAssignments[0]);
+    }
+
+    /**
+     * @covers \eZ\Publish\API\Repository\RoleService::countRoleAssignments()
+     */
+    public function testCountRoleAssignmentsWithDeletedUser(): void
+    {
+        $repository = $this->getRepository();
+        $roleService = $repository->getRoleService();
+        $userService = $repository->getUserService();
+
+        $role = $roleService->loadRoleByIdentifier('Editor');
+        $roleAssignments = $roleService->loadRoleAssignments($role);
+        $expectedCount = count($roleAssignments);
+
+        // Adding user should add '1' to the assignments count
+        $newUser = $this->createUser('login', 'Test', 'Test');
+        $roleService->assignRoleToUser($role, $newUser);
+        $expectedCount = $expectedCount + 1;
+
+        $roleAssignments = $roleService->loadRoleAssignments($role);
+
+        self::assertEquals($expectedCount, count($roleAssignments));
```

Comment:
On a closer look, I see that you're actually not testing here countRoleAssignment method, but still `loadRoleAssignments`. This is good, we need more coverage for deleting user and unassigning role. Here just test method name needs adjustment. Reflected the change in #374